pkg/image/validation/kubernetes.go

```diff
@@ -117,6 +113,48 @@ func validateNodes(k8s *image.Kubernetes) []FailedValidation {
 	return failures
 }
 
+func validateNetwork(k8s *image.Kubernetes) []FailedValidation {
+	var failures []FailedValidation
+	ip := k8s.Network.APIVIP
+
+	numNodes := len(k8s.Nodes)
+	if numNodes <= 1 {
+		// Single node cluster, node configurations are not required
+		return failures
+	}
+
+	if ip == "" {
+		failures = append(failures, FailedValidation{
+			UserMessage: "The 'apiVIP' field is required in the 'network' section when defining entries under 'nodes'.",
+		})
+	}
+
+	if ip != "" {
+		parsedIP, err := netip.ParseAddr(ip)
```

Comment:
Nit: This shows really well why extracting a new variable for something that's easily readable (`k8s.NetworkAPIVIP`) is suboptimal. Now we have to define `parsedIP` instead of using `ip` here which would be simpler, imo.

Reply:
A lot of these instances make much more sense in the context of dual-stack handling and I just kept the format while making it pick one or the other but I'll clean everything up.